### PR TITLE
Add option to infer boot type and mode from template

### PR DESCRIPTION
**Pull request by davidjumani**
### Description

Fixes https://github.com/apache/cloudstack/issues/5932

- Adds the ability to set the boot type and mode in the template itself
- Infers the boot type and mode from the template details in the DeployVM form

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Add the UEFI : SECURE / LEGACY setting to the template :
Screenshot from 2022-02-10 11-02-53

Screenshot from 2022-02-10 11-03-10

#### Watch it inferred in the DeployVM form :

Screenshot from 2022-02-10 11-04-35

### CODE REVIEW

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan ui

**Review by DaanHoogland — APPROVED** [review]
cltgm

**Review by utchoang — APPROVED** [review]
UI LGTM!

**Comment by DaanHoogland** [comment]
@davidjumani any backend tsting that needs to be done for this?

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by shwstppr — COMMENTED** [review]
@davidjumani Looks good but maybe it should mention that this will work only for UI and if using API directly user will still need to provide boot options?

**Comment by nvazquez** [comment]
@davidjumani can you please rebase main branch and target it to main?

**Comment by davidjumani** [comment]
@shwstppr Thanks, Good catch. I think this issue will be handled as part of https://github.com/apache/cloudstack/issues/5913 for deploy-as-is and regular templates

**Comment by davidjumani** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Review by nvazquez — APPROVED** [review]
LGTM

File: ui/src/views/compute/DeployVM.vue [hunk 1]
```diff
@@ -1442,6 +1444,9 @@ export default {
         if (template) {
           var size = template.size / (1024 * 1024 * 1024) || 0 // bytes to GB
           this.dataPreFill.minrootdisksize = Math.ceil(size)
+          this.defaultBootType = this.template?.details?.UEFI ? 'UEFI' : ''
+          this.fetchBootModes(this.defaultBootType)
+          this.defaultBootMode = this.template?.details?.UEFI
```
Comment (rohityadavcloud) [inline ui/src/views/compute/DeployVM.vue:1449]: Why this is declared twice?

Comment (davidjumani) [reply ui/src/views/compute/DeployVM.vue:1449]: The boot type is determined if the `UEFI` key is present in the template details
the boot mode is the value of the key if present

**Pull request merged**